packages/zoe/src/contracts/multipoolAutoswap/swap.js

```diff
@@ -223,18 +232,18 @@ export const makeMakeSwapInvitation = (
         ),
         Central: centralAmountMath.subtract(
           brandInPool.getCentralAmount(),
-          centralAmount,
+          improvedCentralAmount,
         ),
       });
 
       const poolBrandOutStaging = brandOutPool.getPoolSeat().stage({
         Central: centralAmountMath.add(
           brandOutPool.getCentralAmount(),
-          centralAmount,
+          improvedCentralAmount,
         ),
         Secondary: brandOutAmountMath.subtract(
           brandOutPool.getSecondaryAmount(),
-          amountOut,
+          improvedAmountOut,
         ),
       });
 
```

Comment:
I think all changes to this file should be reverted, as we should take the user's `give` as a statement of what they actually want to give, and not second-guess them.